src/e2ee.rs

```diff
@@ -343,16 +336,16 @@ impl E2eeHelper {
                     if let Some(ref mut peerstate) = peerstate {
                         if let Some(ref header) = autocryptheader {
                             peerstate.apply_header(&header, message_time);
-                            peerstate.save_to_db(&context.sql, false);
+                            peerstate.save_to_db(&context.sql, false).unwrap();
```

Comment:
Just as above, these are new panics being introduced.  where before failure to write to the db was silently ignored.  I think this unwise and while I don't expect ultimately perfect code I'd choose one of:
1. Keep silently ignoring, e.g. `.ok()`
2. Bail, e.g. `?`
And optionally with either: log a warning.

Same below, basically search for all `.save_to_db()` calls.